### ISSUE TITLE
Add support for `do yeet`

### ISSUE THREAD

**Issue by mili-l** [intellij-rust/intellij-rust#10347]
## Problem description
Let's support for [yeet_expr](https://doc.rust-lang.org/beta/unstable-book/language-features/yeet-expr.html) feature

```rust
#![feature(yeet_expr)]

fn foo() -> Result<String, i32> {
    do yeet 4;
}
assert_eq!(foo(), Err(4));

fn bar() -> Option<String> {
    do yeet;
}
assert_eq!(bar(), None);
```

**Comment by afetisov** [comment]
I think it would be nice to support experimental syntax at least at the parser level, so that it doesn't result in spurious errors. After all, wouldn't we want the plugin to be more useful for working on compiler development and experimental features?
